helmctlr/remote_chart.go

```diff
@@ -0,0 +1,96 @@
+package helmctlr
+
+import (
+	"strings"
+
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"errors"
+
+	"encoding/base64"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/helm/pkg/downloader"
+	"k8s.io/helm/pkg/getter"
+	"k8s.io/helm/pkg/helm/environment"
+	"k8s.io/helm/pkg/helm/helmpath"
+)
+
+// getChartRef reads the chart entry from CR spec and returns chart entry or empty string.
+func getChartRef(r *unstructured.Unstructured) string {
+	if spec, ok := r.Object["spec"].(map[string]interface{}); ok {
+		if chart, ok := spec["chart"].(string); ok {
+			return chart
+		}
+	}
+
+	return ""
+}
+
+// Gets chart name and chart version from the passed chartRef, downloads the chart from repo,
+// puts it into versioned folder, and returns that folder
+// Uses chart downloader to download the charts
+// 	- chart downloader uses the version passed, but if version is empty, pulls the latest version.
+// Prerequisite: Repo should have been initialized under HELM_HOME
+func getRemoteChart(chartRef string) (string, error) {
```

Comment:
Can we add more tests to handle the error cases.

Reply:
@nraddy Codecov report highlights the cases that need unit testing.

https://codecov.io/gh/wpengine/lostromos/pull/66/diff?src=pr&el=tree#diff-aGVsbWN0bHIvcmVtb3RlX2NoYXJ0Lmdv